### ISSUE TITLE
deppart: Crash when using old-style APIs with very large regions

### ISSUE THREAD

**Issue by elliottslaughter** [StanfordLegion/legion#288]
It appears that the shims supporting the old-style APIs on the `deppart` branch have a space complexity of O(number of elements), as opposed to say O(number of ranges in coloring). This means that users of the old APIs may run out of memory even when being careful to construct ranges properly.

To reproduce:

Copy `language/examples/pennant.tests` from the `spmd-sc17-master` branch to `language/examples/spmd.pennant.tests` (and re-checkout `deppart`).

```
./regent.py examples/pennant_fast.rg examples/spmd.pennant.tests/leblanc_long1024x30/leblanc.pnt -fflow-spmd 1 -fvectorize-unsafe 1 -npieces 100 -numpcx 1 -numpcy 100 -seq_init 0 -par_init 1 -fflow-spmd-shardsize 100 -level inst=1 -logfile inst_%.log
```

With a slight tweak to `inst_impl.cc`, this generates the following instances:

```
[0 - 7fbeac26d700] {2}{inst}: creating new local instance: 6000000000000000
[0 - 7fbeac26d700] {1}{inst}: instance layout: inst=0 layout=Layout(bytes=30198988800, align=32, fields={1=0+0, 2=1+0}, lists=[[<0>..<1887436799>->affine(<8>+0)], [<0>..<1887436799>->affine(<8>+15099494400)]])
[0 - 7fbeac26d700] {1}{inst}: allocation completed: inst=6000000000000000 offset=0
```

**Comment by lightsighter** [comment]
This should be fixed with da885b3 and a533fcc.  Assigning back to @elliottslaughter to confirm.

**Comment by elliottslaughter** [comment]
Confirmed.
